### DISCUSSION TITLE
Ableton 11 unable to delete Undo history IF audiogridder was used in a session

### DISCUSSION THREAD

**Discussion by dhstsw** [apohl79/audiogridder#539] [Help]
As per title.
If i load an instance of AG in Live, when i close the program (removing the AG instance or not before closing doesn't make any difference), Live pops up the 2 following info windows (one after the other):
![image](https://user-images.githubusercontent.com/59790194/117277707-c6cbaa00-ae68-11eb-8ccb-db27d5c9a3d3.png)
![image](https://user-images.githubusercontent.com/59790194/117277750-cf23e500-ae68-11eb-9b9d-1ee6a5627016.png)

This does happen *only* if AG was used.

Plugin log attached (no core files produced), but i don't see anything weird in it.
I think this started with 1.2.0
[AudioGridderPlugin_2021-05-06_12-38-37.zip](https://github.com/apohl79/audiogridder/files/6433488/AudioGridderPlugin_2021-05-06_12-38-37.zip)

**Comment by apohl79** [comment]
tbh, no idea what live is doing :-)

1) Does it work again when you start live without loading an AG plugin and closing it?
2) Can you check the file in the first dialog? Maybe the permissions changed on the file, and live just can't delete it, but I would be wondering if (1) would be true in this case
3) Can you start live, check the file, load AG and see if something on the file changes?

I can't test all of these things myself. The log shows nothing suspicious and I don't know how AG could trigger this behaviour.

**Reply by mgfor** [reply]
Not sure if you noticed my answer. The recently introduced tray icon seems to be the problem when closing Live. 
When I disable the tray icon of AG, Live does not show the error anymore.

**Comment by dhstsw** [comment]
> Does it work again when you start live without loading an AG plugin and closing it?

Yup. 

> Can you check the file in the first dialog? Maybe the permissions changed on the file, and live just can't delete it, but I would be wondering if (1) would be true in this case

Basically, when you start Live it  creates that Undo folder and puts 2 files in it (wich i guess help it to keep track of the whole 'undo situation'). When exiting the DAW, il clears it content. No problem in all situations. The only problem arises when i load an instance of AG.

> Can you start live, check the file, load AG and see if something on the file changes?

The file changes (it's named "0.band"), but that happens with any plugin. What matters is that the attributes (and/or ownership) of the file(s) don't.

> I can't test all of these things myself. The log shows nothing suspicious and I don't know how AG could trigger this behaviour.

I started noticing it with the beta of 1.2 but i guess it would be an hell of a work to understand it.
Well, if you'll ever have the time to try, i think the demo of Live is free and fully functional (save excluded).

Thanks for the great work! :)

C.

BTW: i guess it won't help, but i'll attach 2 zip here. 
The 1st one contains the "virgin" files (Live just loaded).
The second one contains the files after loading AG.
[1.zip](https://github.com/apohl79/audiogridder/files/6436645/1.zip)
[2.zip](https://github.com/apohl79/audiogridder/files/6436648/2.zip)

BTW2: i also noticed that if i remove AG from the midi track and *replace* it with another plugin (i'm talking about instruments here, didn't try with the FX or Midi versions), then Live exists without problems.

**Reply by apohl79** [reply]
Ok looks like the 0.band file contains the history actions including the plugin state as well as the plugin parameters as it seems. So maybe live has some issue with the automation changes, no idea. That said, I tested the demo on macOS and I see the same file being created, but no issues on close. I also tried with 32 up to 512 parameters (plugin config "NumberOfAutomationSlots" setting), but it does not complain.

**Reply by dhstsw** [reply]
Yup, i guess it's unrelated to the number of parameters of the plugin, must be something else.
Would be nice to have someone running Live on Windows to confirm or unconfirm this behaviour.
Anyway, thanks for looking into it.
I'm sure that eventually the situation will solve itself :)
C.

**Reply by dhstsw** [reply]
Interesting: i just noticed the problem is there only with the VST3 version of the plugin. With VST2 Ableton exists without problems.

**Reply by fallynnknivez** [reply]
I am running windows 10, ableton live 11.0.6 and having this issue. I am brand new to audiogridder (as of yesterday).

I originally was ignoring the error and came here because the plugins were not being saved when i exited and reloaded ableton (the AG plugin would be enpty). I initially didnt think the issues to be related.

anyways, after reading this i tried with the vst2 version of the plugin and it saves, but still throws the undo.cfg error on exit, w/ an audiogridder plugin on a track (no issues if i remove all instances of audiogridder).

i did try reseting permissions and ownership in my ableton documents directory, thinking that maybe there was a permissions issue with the file itself, but no joy. I am hesitant to go changing ownership anywhere else as windows starts to have a nutty when its not the owner of specific files and folders.

edit: i am running the beta version of audiogridder. When i get home i plan to swap both machines over to the other version and see if issue persists.

Update: 1.1.1 does not throw the undo.cfg error, but the vst3 still does not save upon closing ableton. (when you reopen the project the audiogridder plugin is empty), the vst2 seems to work fine in all respects, unless, when you open the plugin, you have to click to see the gui, whereas with 1.2 it was just open already, is an issue? (not sure if this is just a thing with the older version).

**Reply by tedgrm** [reply]
I am seeing the same behaviour as @fallynnknivez in Ableton 11.0.6 on Windows 11.

**Reply by mgfor** [reply]
I have the same problem. Disabling the tray icon on audiogridder resolved this.
